src/Analyzers/Core/CodeFixes/MakeDeclarationPartial/AbstractMakePartialCodeFixProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.MakeDeclarationPartial
+{
+    internal abstract class AbstractMakeDeclarationPartialCodeFixProvider<TDeclarationSyntax> : CodeFixProvider
```

Comment:
@CyrusNajmabadi @mavasani  Do we prefer deriving from `SyntaxEditorBasedCodeFixProvider`

Reply:
yes.

Reply:
Seems reasonable, but there is one slight problem: I can implement fix all support, but I think `ContainingMember` and `ContainingType` scopes are not needed here since the fix is about type declarations. `SyntaxEditorBasedCodeFixProvider` don't currently have an ability to override supported `FixAllScope`s. Is it ok if I add such an ability right in this PR (because it's an `internal` buddy), or I still need to fill an issue, pass API validation etc.?

Reply:
@CyrusNajmabadi ?